src/main/java/com/google/codeu/data/Datastore.java

```diff
@@ -81,4 +81,33 @@ public void storeMessage(Message message) {
 
     return messages;
   }
+
+    /** Stores the User in Datastore. */
+    public void storeUser(User user) {
+        Entity userEntity = new Entity("User", user.getEmail());
+        userEntity.setProperty("email", user.getEmail());
+        userEntity.setProperty("aboutMe", user.getAboutMe());
+        datastore.put(userEntity);
+    }
+
+    /**
+     * Returns the User owned by the email address, or
+     * null if no matching User was found.
```

Comment:
It is best to make the possibility of null-ness/invalidity a part of the signature. To do so, switch from `User` as the return type to `Optional<User>`.

Reply:
@michaelsafyan with using optional, should we be using Try-Catch blocks with it? Or is that redundant? 
I was thinking of having the try and catch blocks to log where we had a nullcase.

Reply:
In terms of nullness testing (or any similar checking for that matter), the key thing to guarantee is that errors are thrown from the true source of the error, not at a later time as a result of propagated errors / invalid state.

If you use an object in the course of a function where that object's validity will be implicitly tested as a result of the use, there is no need to explicitly check the validity. For example, if you have code like:

       void doSomething(Optional<Foo> fooOrAbsent) {
             // ...
             Foo foo = fooOrAbsent.get();
             // ...
       }

In the above, there is no need to check that `fooOrAbsent` is non-null; calling `.get()` will throw `NullPointerException` (the same behavior that proper null testing that you do manually should similarly do).

If, however, you store a value that you access later in a different function as in:

       MyConstructor(Optional<Foo> fooOrAbsent) {
          // ...
          this.fooOrAbsent = fooOrAbsent;
          // ...
       }

        void doSomething() {
           // ... use fooOrAbsent here ...
        }

In a case like the above, you should validate the data before storing it. This way, you ensure the class invariant (i.e. you ensure that the state of the class is valid), and you ensure that errors are propagated in the correct location (to the code that invoked the constructor improperly, not the code that correctly invoked "doSomething").